### PR TITLE
Fix functional tests on staging

### PR DESCRIPTION
**Pull request by willbarton**
This PR makes a couple of separate changes intended to fix our functional test runs on staging:

1. Fixes the data-option lookup for categories in the filterable list blog tests. This change is to how we look up the `data-option` from human-friend strings by converting the human-friendly string to kebab-case. 
   
   The issue was, “Data, Research, and Policy” was getting converted to `data,-research,-and-policy`, when the actual `data-option` value is `data-research-policy`. 

   The fix here is wrapped in a utility function to perform this formatting by using a regex to drop words (“and”) and characters (“,”, “’”) before kebab-casing the resulting string.
   
   I’ve made use of this function in all the places where we lookup categories and topics from human-friendly strings.

2. I've added [`@cypress/skip-test`](https://github.com/cypress-io/cypress-skip-test) to let us conditionally skip tests in different environments, and applied that to the FIG tests. Currently they'll skip when running on staging.

@wpears @contolini Can you confirm I've added the `skip-test` package correctly? @contolini do you have any concerns with how I've skipped the FIG tests on staging? 

I've also added the `CYPRESS_ENVIRONMENT=github` to our functional test GHA workflow so that we could potentially transition to `skipOn('github')` in place of our big list of specs, but I have not made that change here.

## How to test this PR

To run the tests against staging with the `CYPRESS_ENVIRONMENT`: 

```
CYPRESS_ENVIRONMENT=staging yarn cypress run --config baseUrl=https://[STAGING_URL]/
```

All the tests should now pass.

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)

### CODE REVIEW

File: test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.js [hunk 1]
```diff
@@ -1,207 +1,221 @@
...
 const fig = new FilingInstructionGuide();
 
-describe( '1071 Filing Instruction Guide (FIG)', () => {
+skipOn( 'staging' , () => {
+
+  describe( 'testing', () => {
+
+    it('should print cypress environment', () => {
+      console.log(Cypress.env());
+      console.log(Cypress.env('ENVIRONMENT'));
```
Comment (contolini) [inline test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.js:12]:
```suggestion
      cy.log(Cypress.env());
      cy.log(Cypress.env('ENVIRONMENT'));
```

**Review by contolini — APPROVED** [review]
Thanks for this fix! It makes me wonder: for things like the FIG, should we be adding logic that skips tests based on the existence of the test page *instead* of relying on environment variables? e.g. `skipOn( figUrlReturns404() )`?

File: test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.js (hunk 1 above)
Comment (willbarton) [reply test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.js]: Whoops. That was totally not intended for commit.

**Comment by willbarton** [comment]
> Thanks for this fix! It makes me wonder: for things like the FIG, should we be adding logic that skips tests based on the existence of the test page *instead* of relying on environment variables? e.g. `skipOn( figUrlReturns404() )`?

I do like this idea, I think. We wouldn't want to accidentally skip a set of tests if, like, a page gets moved or removed, but I think it'd be a good pattern for new work or feature-flagged things.

**Pull request merged**